pulsar-proxy/src/main/java/org/apache/pulsar/proxy/server/ProxyConfiguration.java

```diff
@@ -43,109 +46,325 @@
 public class ProxyConfiguration implements PulsarConfiguration {
     private final static Logger log = LoggerFactory.getLogger(ProxyConfiguration.class);
 
-    // Local-Zookeeper quorum connection string
+    @Category
+    private static final String CATEGORY_SERVER = "Server";
+    @Category
+    private static final String CATEGORY_BROKER_DISCOVERY = "Broker Discovery";
+    @Category
+    private static final String CATEGORY_AUTHENTICATION = "Proxy Authentication";
+    @Category
+    private static final String CATEGORY_AUTHORIZATION = "Proxy Authorization";
+    @Category(
+        description = "the settings are for configuring how proxies authenticates with Pulsar brokers"
+    )
+    private static final String CATEGORY_CLIENT_AUTHENTICATION = "Broker Client Authorization";
+    @Category
+    private static final String CATEGORY_RATE_LIMITING = "RateLimiting";
+    @Category
+    private static final String CATEGORY_TLS = "TLS";
+    @Category
+    private static final String CATEGORY_TOKEN_AUTH = "Token Authentication Provider";
+    @Category
+    private static final String CATEGORY_HTTP = "HTTP";
+
+    @FieldContext(
+        category = CATEGORY_BROKER_DISCOVERY,
+        doc = "The ZooKeeper quorum connection string (as a comma-separated list)"
+    )
     private String zookeeperServers;
+    @FieldContext(
+        category = CATEGORY_BROKER_DISCOVERY,
+        doc = "Configuration store connection string (as a comma-separated list)"
+    )
+    private String configurationStoreServers;
+    @FieldContext(
+        category = CATEGORY_BROKER_DISCOVERY,
+        doc = "Global ZooKeeper quorum connection string (as a comma-separated list)"
+    )
     @Deprecated
-    // Global-Zookeeper quorum connection string
     private String globalZookeeperServers;
 
-    // Configuration Store connection string
-    private String configurationStoreServers;
-
-    // ZooKeeper session timeout
+    @FieldContext(
+        category = CATEGORY_BROKER_DISCOVERY,
+        doc = "ZooKeeper session timeout (in milliseconds)"
+    )
     private int zookeeperSessionTimeoutMs = 30_000;
 
     // if Service Discovery is Disabled this url should point to the discovery service provider.
```

Comment:
Should we also removed these comments (as they may diverge over time with the doc string) ?

Reply:
removed those comments